metaflow/plugins/kfp/kfp_decorator.py

```diff
@@ -87,3 +82,14 @@ def task_finished(
                     )
                     url = urlparse(path)
                     s3.upload_fileobj(f, url.netloc, url.path.lstrip("/"))
+        else:
+            # we are publishing to a Metadata service
+            pass
+
+        # TODO: If this step is retried, then S3 doesn't guarantee consistency that
+        #  the last uploaded data is returned. It'd be very rare that the data would change
+        #  as this is the last step, and we assume the same inputs should produce the same
+        #  outputs.
+        context_dict: Dict[str, str] = self.split_contexts.build_context_dict(flow)
+        KfpSplitContext.save_context_to_local_fs(context_dict)
+        self.split_contexts.upload_context_to_flow_root(context_dict)
```

Comment:
Wondering if we can make a higher-level method here on `KfpSplitContext` to encapsulate all these operations so here would just be a single command to "finish" off that step, something like `self.split_contexts.save(flow)` and it could optionally return a `context_dict` but under the hood it'd just call these operations just to encapsulate this a little nicer. Thoughts?